core/tests/test_utils.py

```diff
@@ -331,41 +320,294 @@ def set_multi(self, key_value_mapping):
             bool. Whether the set action succeeded.
         """
         assert isinstance(key_value_mapping, dict)
-        for key, value in key_value_mapping.items():
-            self.cache_dict[key] = value
+        self._CACHE_DICT.update(key_value_mapping)
         return True
 
     def delete_multi(self, keys):
         """Deletes multiple keys in the cache dictionary.
 
         Args:
-            keys: list(str). The keys (strings) to delete.
+            keys: list(str). The keys to delete.
 
         Returns:
             int. Number of successfully deleted keys.
         """
-        number_of_deleted_keys = 0
-        for key in keys:
-            assert isinstance(key, python_utils.BASESTRING)
-            if key in self.cache_dict:
-                del self.cache_dict[key]
-                number_of_deleted_keys += 1
-        return number_of_deleted_keys
+        assert all(isinstance(key, python_utils.BASESTRING) for key in keys)
+        keys_to_delete = [key for key in keys if key in self._CACHE_DICT]
+        for key in keys_to_delete:
+            del self._CACHE_DICT[key]
+        return len(keys_to_delete)
 
 
 class TestBase(unittest.TestCase):
```

Comment:
Chunk of code deleted in this class have been moved to AppEngineTestBase. They were dependent on AppEngineTestBase.setUp to begin with, so all of this logic was a bug waiting to happen.